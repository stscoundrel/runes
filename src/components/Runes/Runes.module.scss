.section {
  min-height: calc(100vh - 100px);
  overflow: scroll;
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.header {
  text-align: center;
}

.runic {
  font-size: rem(26px);
  letter-spacing: rem(1.5px);
  color: $color-dark;
  word-break: break-all;
}