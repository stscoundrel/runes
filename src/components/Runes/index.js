import { useState } from 'react'

// Rune adapters.
import youngerFuthark from 'younger-futhark'
import elderFuthark from 'elder-futhark'

import styles from './Runes.module.scss'

export default function Runes() {
  const [content, setContent] = useState('')

  return (
    <section className={styles.section}>
      <header className={styles.header}>
        <h1>Runes</h1>
        <p>Translate text to runic script</p>
      </header>

      <label htmlFor="transcribe">Text to transcribe</label>
      <textarea name="transcribe" value={content} onChange={(e) => setContent(e.target.value)} />

      <p>Younger Futhark</p>
      <p className={styles.runic}>{youngerFuthark.lettersToRunes(content)}</p>

      <p>Elder Futhark</p>
      <p className={styles.runic}>{elderFuthark.lettersToRunes(content)}</p>

    </section>
  )
}
