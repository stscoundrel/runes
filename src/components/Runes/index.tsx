import { useState } from 'react'

// Rune adapters.
import youngerFuthark from 'younger-futhark'
import elderFuthark from 'elder-futhark'
import futhorc from 'futhorc'
import futhork from 'futhork'
import stavelessFuthark from 'staveless-futhark'

import styles from './Runes.module.scss'

export default function Runes() {
  const [content, setContent] = useState('')

  return (
    <section className={styles.section}>
      <header className={styles.header}>
        <h1>Runes</h1>
        <p>Translate text to runic script</p>
      </header>

      <label htmlFor="transcribe">Text to transcribe</label>
      <textarea id="transcribe" value={content} onChange={(e) => setContent(e.target.value)} />

      <p>Younger Futhark</p>
      <small>Long branch</small>
      <p className={styles.runic}>{youngerFuthark.lettersToLongBranchRunes(content)}</p>

      <small>Short twig</small>
      <p className={styles.runic}>{youngerFuthark.lettersToShortTwigRunes(content)}</p>

      <p>Elder Futhark</p>
      <p className={styles.runic}>{elderFuthark.lettersToRunes(content)}</p>

      <p>Futhorc</p>
      <p className={styles.runic}>{futhorc.lettersToRunes(content)}</p>

      <p>Futhork</p>
      <p className={styles.runic}>{futhork.lettersToRunes(content)}</p>

      <p>Staveless Futhark</p>
      <p className={styles.runic}>{stavelessFuthark.lettersToRunes(content)}</p>

    </section>
  )
}
