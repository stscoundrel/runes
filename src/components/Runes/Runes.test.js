import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Runes from './index'

describe('Runes component', () => {
  test('Does not crash', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Runes />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  test('Matches snapshot', () => {
    const tree = renderer.create(<Runes />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
