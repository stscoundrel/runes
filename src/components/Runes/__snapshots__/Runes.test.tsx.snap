// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Runes component Matches snapshot 1`] = `
<section
  className="section"
>
  <header
    className="header"
  >
    <h1>
      Runes
    </h1>
    <p>
      Translate text to runic script
    </p>
  </header>
  <label
    htmlFor="transcribe"
  >
    Text to transcribe
  </label>
  <textarea
    id="transcribe"
    onChange={[Function]}
    value=""
  />
  <p>
    Younger Futhark
  </p>
  <small>
    Long branch
  </small>
  <p
    className="runic"
  />
  <small>
    Short twig
  </small>
  <p
    className="runic"
  />
  <p>
    Elder Futhark
  </p>
  <p
    className="runic"
  />
  <p>
    Futhorc
  </p>
  <p
    className="runic"
  />
  <p>
    Futhork
  </p>
  <p
    className="runic"
  />
  <p>
    Staveless Futhark
  </p>
  <p
    className="runic"
  />
</section>
`;

exports[`Runes component Typing text to input produces transcriptions 1`] = `
<section
  className="section"
>
  <header
    className="header"
  >
    <h1>
      Runes
    </h1>
    <p>
      Translate text to runic script
    </p>
  </header>
  <label
    htmlFor="transcribe"
  >
    Text to transcribe
  </label>
  <textarea
    id="transcribe"
    onChange={[Function]}
    value="Some text"
  />
  <p>
    Younger Futhark
  </p>
  <small>
    Long branch
  </small>
  <p
    className="runic"
  >
    ᛋᚢᛘᛁ:ᛏᛁᛋᛏ
  </p>
  <small>
    Short twig
  </small>
  <p
    className="runic"
  >
    ᛌᚢᛘᛁ:ᛐᛁᛌᛐ
  </p>
  <p>
    Elder Futhark
  </p>
  <p
    className="runic"
  >
    ᛋᛟᛗᛖ:ᛏᛖᛋᛏ
  </p>
  <p>
    Futhorc
  </p>
  <p
    className="runic"
  >
    ᛋᚩᛗᛖ:ᛏᛖᛉᛏ
  </p>
  <p>
    Futhork
  </p>
  <p
    className="runic"
  >
    ᛋᚮᛘᚽ:ᛏᚽᛋᛏ
  </p>
  <p>
    Staveless Futhark
  </p>
  <p
    className="runic"
  >
    ╵ˎ⠃ᛁ:⸍ᛁ╵⸍
  </p>
</section>
`;
